### ISSUE TITLE
[java] False positive in UnusedAssignment when assignment is in conditional statement

### ISSUE THREAD

**Issue by zbynek** [pmd/pmd#5940]
**Affects PMD Version:** 
6.55.0, 7.16.0

**Rule:**
[UnusedAssignment](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedassignment)

**Description:**

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public void process() {
	boolean ok2 = true;
	if (false && (ok2 = false)) {
		return;
	}
	System.out.println(ok2);
    }
}

```
Since the second assignment of `ok2` is never reached, the first one is not redundant.

**Expected outcome:**

PMD reports a violation at line 3 but that's wrong. That's a false positive.

**Running PMD through:** Gradle
